lua/nvim-navic/lib.lua

```diff
@@ -254,24 +254,15 @@ function M.request_symbol(for_buf, handler, client_id, file_uri)
 		}
 	end
 
-	vim.lsp.buf_request_all(
-		for_buf,
-		"textDocument/documentSymbol",
-		{ textDocument = textDocument_argument },
-		function(symbols)
-			if symbols[client_id] == nil then
-				return
-			elseif symbols[client_id].error ~= nil or symbols[client_id].result == nil then
-				vim.defer_fn(function()
-					M.request_symbol(for_buf, handler, client_id)
-				end, 750)
-			elseif symbols[client_id].result ~= nil then
-				if vim.api.nvim_buf_is_valid(for_buf) then
-					handler(for_buf, symbols[client_id].result)
-				end
+	client.request("textDocument/documentSymbol", { textDocument = textDocument_argument }, function(err, symbols, _, _)
```

Comment:
Nice! I didn't know we could request for the individual client like this. I always thought it was weird that there is only `buf_request_all` function and no `buf_request` for specific client. Good work!